engine/script/src/test/test_script_msg.cpp

```diff
@@ -388,23 +388,46 @@ TEST_F(ScriptMsgTest, TestURLToString)
     ASSERT_EQ(dmMessage::RESULT_OK, dmMessage::NewSocket("socket", &socket));
     ASSERT_EQ(dmMessage::RESULT_OK, dmMessage::NewSocket("very_very_very_very_very_very_very_very_very_very_very_very_socket", &overflow_socket));
     ASSERT_TRUE(dmScriptTest::RunString(L,
-        "local url = msg.url()\n"
-        "print(tostring(url))\n"
-        "url = msg.url(\"socket\", \"path\", \"test\")\n"
-        "print(tostring(url))\n"
-        "-- overflow\n"
-        "url = msg.url(\"very_very_very_very_very_very_very_very_very_very_very_very_socket\", \"path\", \"test\")\n"
-        "print(tostring(url))\n"
+        "test_msg_fn = function (url, expected)\n"
```

Comment:
I'm still not certain why my tests failed, other than that I got trashed strings.
This was also backed up by the fact that the CI failed/succeeded randomly for the same static testdata.

So it was definitely misbehaving, and I think we actually need to change the reversehash back to the way it was, and handle it locally where the feature was wanted.